homeassistant/core.py

```diff
@@ -368,10 +367,6 @@ def async_fire(self, event_type: str, event_data=None,
 
         This method must be run in the event loop.
         """
-        if event_type != EVENT_HOMEASSISTANT_STOP and \
```

Comment:
This was silently (because exception suppressed) killing shutdown tasks (noticed it when adding shutdown event in automation). It should be fine to just wait till all is done. Timer is already stopped so nothing new should come in.